db/warehouse/migrate/20250516141401_add_step_available_at.rb

```diff
@@ -0,0 +1,27 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class AddStepAvailableAt < ActiveRecord::Migration[7.1]
+  def up
+    add_column :wfe_steps, :available_at, :datetime, null: true
+
+    safety_assured do
+      execute <<~SQL
+        UPDATE wfe_steps SET available_at = created_at
+      SQL
+
+      change_column_null :wfe_steps, :available_at, false
+
+      # Index because we have to sort steps on this field
+      add_index :wfe_steps, :available_at
```

Comment:
I'd be glad for feedback on this index. It seemed worthwhile to me, given that we know we will be sorting on the field. But let me know if there are some additional considerations that I should take into account. Maybe our general practice is to not over-optimize on db indices until we actually see a need for them?

Reply:
When you're referring to db sorting on this column, is it just happening when we resolve `current_steps` on Referral? I'm not seeing anywhere else but maybe I'm missing something. That should only return a handful of steps max so maybe we should just sort them in memory. I can't imagine we ever have a need to sort a large number of steps by availability? (Since we only really care about that per-referral, and each referral only has a handful  of steps).

Reply:
By doing it in memory, I mean `load_ar_association(object, :current_steps).sort_by(&:available_at)`

Reply:
Yes that's what I am referring to. Sounds good, I can remove the index and sort in memory